packages/devextreme/testing/tests/DevExpress.ui.widgets/ganttParts/milestone.tests.js

```diff
@@ -0,0 +1,114 @@
+                assert.true(title.hasClass(Consts.TASK_TITLE_SELECTOR.slice(1)), 'taskHTML has task title');
+                assert.true(milestone.hasClass(Consts.MILESTONE_SELECTOR.slice(1)), 'taskHTML has milestone');
+                assert.equal(item.taskHTML.length, 2, 'taskHTML has 2 children');
+
+                title.appendTo(customHtml);
+                milestone.appendTo(customHtml);
+            }
+            return item.taskHTML;
+        };
+        const options = {
+            tasks: { dataSource: customTasks },
+            taskContentTemplate: customTaskFunction
+        };
+        this.createInstance(options);
+        this.clock.tick(10);
+        const milestoneWrapper = this.$element.find(Consts.MILESTONE_WRAPPER_SELECTOR);
+        assert.equal(milestoneWrapper.length, 1, 'milestone wrapper exists');
+        assert.equal(milestoneWrapper.children().length, 1, 'milestone wrapper has 1 child');
+        assert.equal(milestoneWrapper.html(), customHtml[0].outerHTML, 'milestone html matches custom html');
+    });
+
+    test('custom template, custom html', function(assert) {
+        const customHtml = $(document.createElement('div')).text('milestone');
+        const customTasks = [
+            { 'id': 1, 'parentId': 0, 'title': 'Software Development', 'start': new Date('2019-02-21'), 'end': new Date('2019-03-26'), 'progress': 0 },
+            { 'id': 2, 'parentId': 1, 'title': 'Milestone', 'start': new Date('2019-02-21'), 'end': new Date('2019-02-21'), 'progress': 0 },
+        ];
+        const customTaskFunction = (item, container) => {
+            if(item.isMilestone) {
+                return customHtml.text();
+            }
+            return item.taskHTML;
+        };
+        const options = {
+            tasks: { dataSource: customTasks },
+            taskContentTemplate: customTaskFunction
+        };
+        this.createInstance(options);
+        this.clock.tick(10);
+        const milestoneWrapper = this.$element.find(Consts.MILESTONE_WRAPPER_SELECTOR);
+        assert.equal(milestoneWrapper.length, 1, 'milestone wrapper exists');
+        assert.equal(milestoneWrapper.html(), customHtml[0].outerHTML, 'milestone html matches custom html');
+    });
+
+    test('custom template, custom html + task html', function(assert) {
+        const customHtml = $(document.createElement('div')).text('milestone');
+        const customTasks = [
+            { 'id': 1, 'parentId': 0, 'title': 'Software Development', 'start': new Date('2019-02-21'), 'end': new Date('2019-03-26'), 'progress': 0 },
+            { 'id': 2, 'parentId': 1, 'title': 'Milestone', 'start': new Date('2019-02-21'), 'end': new Date('2019-02-21'), 'progress': 0 },
+        ];
+        const customTaskFunction = (item, container) => {
+            if(item.isMilestone) {
+                $(item.taskHTML[1].cloneNode(true)).appendTo(customHtml);
+                return customHtml.html();
```

Comment:
I think in this test we need to add another div (not default from taskHTML) with different width and height  in the customHtml to check the line in your code: result.x -= width / 2;
By the way no test checked this line, add this please.